sunkit_image/enhance.py

```diff
@@ -0,0 +1,122 @@
+"""
+This module contains functions that can be used to enhance the entire solar image.
+"""
+
+import numpy as np
+import scipy.ndimage as ndimage
+
+__all__ = [
+    "mgn",
+]
+
+
+def mgn(
+    data,
+    sigma=[1.25, 2.5, 5, 10, 20, 40],
+    k=0.7,
+    gamma=3.2,
+    h=0.7,
+    weights=None,
+    truncate=3
+):
+
+    """
+    Multi-scale Gaussian normalization.
+
+    Parameters
+    ----------
+    data : `numpy.ndarray`
+        Image to be transformed.
+    sigma : `list`, optional
+        Range of Gaussian widths to transform over.
+        Defaults to ``[1.25, 2.5, 5, 10, 20, 40]``.
```

Comment:
use astropy units where appropriate.  For example, the sigma argument here should be `sigma=[1.25, 2.5, 5, 10, 20, 40]*u.pix`

Reply:
If the image is simply a `ndarray` and not a `Map`, I am not sure that I understand why there should be units here... This would be the only astropy dependence of the function.

Reply:
Yes for a simple `ndarray` why we need units

Reply:
Fair point.